src/python/pants/util/osutil_test.py

```diff
@@ -13,68 +11,62 @@
+def test_alias_normalization() -> None:
+    for normal_os, aliases in OS_ALIASES.items():
+        for alias in aliases:
+            assert normal_os == normalize_os_name(alias)
 
-    def test_keys_in_aliases(self) -> None:
-        for key in OS_ALIASES.keys():
-            self.assertIn(key, known_os_names())
 
-    def test_no_warnings_on_known_names(self) -> None:
-        for name in known_os_names():
-            with self.captured_logging(logging.WARNING) as captured:
-                normalize_os_name(name)
-                self.assertEqual(
-                    0,
-                    len(captured.warnings()),
-                    f"Received unexpected warnings: {captured.warnings()}",
-                )
+def test_keys_in_aliases() -> None:
+    for key in OS_ALIASES.keys():
+        assert key in known_os_names()
 
-    def test_warnings_on_unknown_names(self) -> None:
-        name = "I really hope no one ever names an operating system with this string."
-        with self.captured_logging(logging.WARNING) as captured:
-            normalize_os_name(name)
-            self.assertEqual(
-                1,
-                len(captured.warnings()),
-                f"Expected exactly one warning, but got: {captured.warnings()}",
-            )
 
-    def test_get_closest_mac_host_platform_pair(self) -> None:
-        # Note the gaps in darwin versions.
-        platform_name_map = {
-            ("linux", "x86_64"): ("linux", "x86_64"),
-            ("linux", "amd64"): ("linux", "x86_64"),
-            ("darwin", "10"): ("mac", "10.6"),
-            ("darwin", "13"): ("mac", "10.9"),
-            ("darwin", "14"): ("mac", "10.10"),
-            ("darwin", "16"): ("mac", "10.12"),
-            ("darwin", "17"): ("mac", "10.13"),
-        }
+def test_no_warnings_on_known_names(caplog) -> None:
+    for name in known_os_names():
+        normalize_os_name(name)
+        assert len(caplog.records) == 0
 
-        def get_macos_version(darwin_version: Optional[str]) -> Optional[str]:
-            host, version = get_closest_mac_host_platform_pair(
-                darwin_version, platform_name_map=platform_name_map
-            )
-            if host is not None:
-                self.assertEqual("mac", host)
-            return version
 
-        self.assertEqual("10.13", get_macos_version("19"))
-        self.assertEqual("10.13", get_macos_version("18"))
-        self.assertEqual("10.13", get_macos_version("17"))
-        self.assertEqual("10.12", get_macos_version("16"))
-        self.assertEqual("10.10", get_macos_version("15"))
-        self.assertEqual("10.10", get_macos_version("14"))
-        self.assertEqual("10.9", get_macos_version("13"))
-        self.assertEqual("10.6", get_macos_version("12"))
-        self.assertEqual("10.6", get_macos_version("11"))
-        self.assertEqual("10.6", get_macos_version("10"))
-        self.assertEqual(None, get_macos_version("9"))
+def test_warnings_on_unknown_names(caplog) -> None:
```

Comment:
Now we use caplog.

Reply:
Excellent!

Reply:
Isn't it nice! The only thing it's missing for me is a type hint for `caplog`, hehe. But I'll take this.